rhobs/alerting/data_plane/prometheus.performance_alerts.yaml

```diff
@@ -0,0 +1,97 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: rhtap-performance-alerting
+  labels:
+    tenant: rhtap
+spec:
+  groups:
+    - name: performance_alerts
+      interval: 1m
+      rules:
+        # ETCD alerts
+        - alert: EtcdFsyncLatency
+          expr: |
+            avg_over_time(histogram_quantile(0.99, rate(etcd_disk_wal_fsync_duration_seconds_bucket[2m]))[10m:]) > 0.01
+          for: 10m
+          labels:
+            severity: warning
+          annotations:
+            summary: >-
+              ETCD slow file system synchronization.
+            description: >-
+              10 minutes avg. 99th etcd fsync latency on {{$labels.pod}} higher than 10ms.
+
+        - alert: EtcdCommitLatency
+          expr: |
+            avg_over_time(histogram_quantile(0.99, rate(etcd_disk_backend_commit_duration_seconds_bucket[2m]))[10m:]) > 0.03
+          for: 10m
+          labels:
+            severity: warning
+          annotations:
+            summary: >-
+              ETCD slow writes observed.
+            description: >-
+              10 minutes avg. 99th etcd commit latency on {{$labels.pod}} higher than 30ms.
+
+        - alert: EtcdProposalFailures
+          expr: |
+            rate(etcd_server_proposals_failed_total[1h]) > 0.05
+          for: 15m
+          labels:
+            severity: warning
+          annotations:
+            summary: >-
+              ETCD raft proposal failures.
+            description: >-
+              Etcd high number of failed proposals on pod {{ $labels.pod }}
+
+        - alert: EtcdSlowNetworkRTT
+          expr: |
+            histogram_quantile(0.99, rate(etcd_network_peer_round_trip_time_seconds_bucket[5m])) > 0.1
+          for: 15m
+          labels:
+            severity: warning
+          annotations:
+            summary: >-
+              High RTT latency on ETCD cluster member requests.
+            description: >-
+              99th etcd RTT latency rate on {{$labels.pod}} higher than 0.1
+
+        # KubeAPI Alerts
+        - alert: KubernetesJobSlowCompletion
+          expr: |
+            kube_job_spec_completions - kube_job_status_succeeded - kube_job_status_failed > 0
```

Comment:
What is this metric for? Does it talk about usual kuberneted Jobs?

Reply:
Yes, this is talking about the usual Kubernetes Job. The complete expression returns a value that tells how many jobs have not been completed (i.e no success/failure) and are still executing. So if a job has been running for 10min past, this alert should get triggered. I guess we could also term this alert for Long Running Job.

Reply:
Looking at stone-prd-rh01, found this:
![image](https://github.com/user-attachments/assets/73c8fafa-5c51-4e1a-a974-4f804b0c88f4)
Unfortunately I do not have insight into how long Jobs in Konflux cluster usually are. Anyway, looks good for me!